### DISCUSSION TITLE
internal consistency problem when checking if BBL causes many runs

### DISCUSSION THREAD

**Discussion by Dekker1** [tectonic-typesetting/tectonic#1186] [Q&A] [answered]
I was wondering whether anyone had run into the following issue. When compiling a very simple tex file

`test.tex`
```tex
\documentclass{article}
\usepackage[square,numbers]{natbib}
\bibliographystyle{abbrvnat}

\title{Bibliography}

\begin{document}

\maketitle

\cite{latexcompanion}

\bibliography{sample}

\end{document}
```

with associated `sample.bib`
```bibtex
@book{latexcompanion,
    author    = "Michel Goossens and Frank Mittelbach and Alexander Samarin",
    title     = "The LaTeX Companion",
    year      = "1993",
    publisher = "Addison-Wesley",
    address   = "Reading, Massachusetts"
}
```

Compiling with `tectonic test.tex` (or with similar files and `tectonic -X build`) results in the following trace:
```
Running TeX ...
Running BibTeX on test.aux ...
Rerunning TeX because bibtex was run ...
warning: internal consistency problem when checking if test.bbl changed
Rerunning TeX because "test.bbl" changed ...
warning: internal consistency problem when checking if test.bbl changed
Rerunning TeX because "test.bbl" changed ...
warning: internal consistency problem when checking if test.bbl changed
Rerunning TeX because "test.bbl" changed ...
warning: internal consistency problem when checking if test.bbl changed
Rerunning TeX because "test.bbl" changed ...
warning: internal consistency problem when checking if test.bbl changed
Rerunning TeX because "test.bbl" changed ...
warning: internal consistency problem when checking if test.bbl changed
warning: TeX rerun seems needed, but stopping at 6 passes
Running xdvipdfmx ...
Writing `test.pdf` (16.11 KiB)
Skipped writing 2 intermediate files (use --keep-intermediates to keep them)
```

It seems that somehow the internal detection cannot detect that the `test.bbl` file hasn't changed.
When keeping intermediates, I cannot even see any changes in `test.bbl`, so I'm not sure what the problem is.

Has anyone run into this issue before? Is there a solution to this problem?
A search in discussions and issues revealed that https://github.com/tectonic-typesetting/tectonic/issues/1185 might have the same issue, but the discussion is about a different error.

**Comment by CraftSpider** [comment, marked as answer]
This is a known problem, the associated issue was simply already closed: https://github.com/tectonic-typesetting/tectonic/issues/1150. The fix is already in the codebase, it is simply waiting on the next release to go out to users.

**Reply by Dekker1** [reply]
Thank you. I'm surprised my search didn't find the issue.
